x-pack/plugins/apm/server/lib/metrics/by_agent/java/non_heap_memory/index.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import theme from '@elastic/eui/dist/eui_theme_light.json';
+import { i18n } from '@kbn/i18n';
+import { Setup } from '../../../../helpers/setup_request';
+import { fetch, NonHeapMemoryMetrics } from './fetcher';
+import { ChartBase } from '../../../types';
+import { transformDataToChart } from '../../../transform_metrics_chart';
+
+const chartBase: ChartBase<NonHeapMemoryMetrics> = {
+  title: i18n.translate('xpack.apm.agentMetrics.java.nonHeapMemoryChartTitle', {
+    defaultMessage: 'Non-Heap Memory'
+  }),
+  key: 'non_heap_memory_area_chart',
```

Comment:
Remind me: what is this key for?

Reply:
Right now it's being used for the react key as we map over the list of charts, as opposed to trying to "slugify" the title etc.